icedyno/preprocess/crop.py

```diff
@@ -49,58 +52,75 @@ def run(self) -> None:
                 os.path.join(year_output_dir, pathlib.Path(cdf_filepath).stem)
                 + f"_grid{self.window_size}"
             )
-            if self.center_coordinates != "None":
-                output_filename += (
-                    f"_{self.center_coordinates.replace(' ', '')}center.nc"
-                )
+            if (self.center_latitude != "None") and (self.center_longitude != "None"):
+                output_filename += f"_{str(self.center_latitude).replace('.', ',')}lat_{str(self.center_longitude).replace('.', ',')}lon.nc"
             else:
                 output_filename += ".nc"
 
+            # Don't recompute file if the expected filename in the output folder already exists.
             if os.path.exists(output_filename):
                 print(cdf_filepath, "already on disk, skipping...")
+                continue
+
+            with xr.open_dataset(cdf_filepath, engine="h5netcdf") as ds:
+                # If specified, center the window on the lat/lon coordinates provided. Otherwise, center on middle of grid.
+                if (self.center_latitude != "None") and (
+                    self.center_longitude != "None"
+                ):
+                    # Project the lat/lon coordinates to the polar stereographic coordinates.
+                    x, y = icedyno.preprocess.geolocation.polar_lonlat_to_xy(
+                        longitude=self.center_longitude, latitude=self.center_latitude
+                    )
+                else:
+                    x = np.min(np.abs(ds.x))
+                    y = np.min(np.abs(ds.y))
+
+                window = self.window_size * 1000  # from km to meters
+
+                center_x = np.min(np.abs(ds.x))
```

Comment:
This is the fix I talked about -- rotating the entire netCDF is too large and slow for the ~3k files we have, so cropping the file to a large window around the polar region that should always include the requested window, then rotating, then selecting the final window was the workaround.
It will fail an assert if the requested window was not inside the first window.